common/model-views.cpp

```diff
@@ -2850,6 +2910,69 @@ namespace rs2
         return get_interpolated_layout(results);
     }
 
+    rs2::frame viewer_model::handle_ready_frames(const rect& viewer_rect, ux_window& window, int devices, std::string& error_message)
+    {
+        texture_buffer* texture_frame = nullptr;
+        points p;
+        frame f{}, res{};
+        try
+        {
+            if (ppf.resulting_queue.poll_for_frame(&f))
+            {
+                frameset frames;
+                p = f.as<points>();
+                if (frames = f.as<frameset>())
```

Comment:
the conditions here are confusing...
`f` can't be a `frameset` and `potins` so `if (frames = f.as<frameset>())` then `p` is empty so the `if(!p)` later inside this scope is redundant.

Reply:
I'll put comments inline